test/webgl/uniforms.spec.js

```diff
@@ -155,14 +178,72 @@ test('WebGL#Uniforms pre verify uniforms', t => {
   t.end();
 });
 
-test('WebGL#Uniforms Program construct/delete', t => {
+test('WebGL#Uniforms Program uniform locations', t => {
   const {gl} = fixture;
 
   const program = new Program(gl, {
     vs: VERTEX_SHADER,
     fs: WEBGL1_FRAGMENT_SHADER
   });
-  t.ok(program instanceof Program, 'Program construction successful');
+
+  for (const uniformName in WEBGL1_GOOD_UNIFORMS) {
+    t.ok(program._uniformSetters[uniformName], `Program found uniform setter ${uniformName}`);
+  }
 
   t.end();
 });
+
+const testSetUniform = gl => t => {
+  const program = new Program(gl, {
+    vs: VERTEX_SHADER,
+    fs: WEBGL1_FRAGMENT_SHADER
+  });
+  program.use();
+
+  const uniforms = Object.assign({}, WEBGL1_GOOD_UNIFORMS);
+
+  program.setUniforms(uniforms);
+  t.pass('Program set uniforms successful');
+
+  for (const uniformName in WEBGL1_GOOD_UNIFORMS) {
```

Comment:
Nit: Add comment `// test setting plain arrays`.
Nit: I'd probably make a fresh clone of the uniforms here.